src/components/faceVerification/standalone/hooks/useGiveUpDialog.js

```diff
@@ -0,0 +1,55 @@
+import React, { useCallback, useContext } from 'react'
+import { isUndefined, negate, pickBy } from 'lodash'
+
+import { fireEvent, FV_GIVEUP } from '../../../../lib/analytics/analytics'
+import AsyncStorage from '../../../../lib/utils/asyncStorage'
+import { useDialog } from '../../../../lib/dialog/useDialog'
+import GiveUpDialog from '../components/GiveUpDialog'
+import { FVFlowContext } from '../context/FVFlowContext'
+import useFVRedirect from './useFVRedirect'
+
+const useGiveUpDialog = (navigation, type) => {
+  const { showDialog } = useDialog()
+  const fvRedirect = useFVRedirect()
+  const { isFVFlow } = useContext(FVFlowContext)
+  const { navigate } = navigation
+
+  const onReasonChosen = useCallback(
+    (reason = undefined) => {
+      const data = pickBy({ reason }, negate(isUndefined))
+
+      if (reason && reason !== 'closed') {
+        AsyncStorage.removeItem('hasStartedFV')
+      }
+
+      fireEvent(FV_GIVEUP, { data, surveyType: type })
+
+      if (isFVFlow) {
+        const redirect = () => fvRedirect(false, reason)
+
+        // await before analytics scripts will perform some activity
+        if (window.requestIdleCallback) {
+          window.requestIdleCallback(redirect)
+        } else {
+          window.requestAnimationFrame(redirect)
```

Comment:
I've done detailed research over polyfills and I could suggest this one, it's most popular and have good quality according npm stats https://github.com/aFarkas/requestIdleCallback 

```javascript
import * as idle from 'requestIdleCallback'

// ...
const requestIdle = window.requestIdleCallback || idle.request

requestIdle(redirect)
```

Reply:
done